### ISSUE TITLE
nextPage() throws Exception

### ISSUE THREAD

**Issue by magedan** [guym4c/airtable-client#2]
Hello,

it looks like nextPage() method in RecordListRequest class is broken.

My use case is very simple and straightforward. Please consider following code:
```
        $result = [];

        // collect all records
        $data = $airtable->list($table);
        
        do {
            $result = $result + $data->getRecords();
        } while ($data->nextPage());
```

After the first iteration I get `Return value of Guym4c\Airtable\Request\RecordListRequest::nextPage() must be an instance of Guym4c\Airtable\Request\RecordListRequest, null returned`. 
It should have been expected since method's declaration doesn't consider null as a return value (it should be `?self` I guess):
```
    public function nextPage(): self {

        if (empty($offset)) {
            return null;
        }

        $this->options['query']['offset'] = $this->offset;
        return $this->getResponse();
    }
```

More than that, `$offset` will be always undefined.
I'm not sure what is the correct/expected behavior, otherwise I would prepare a PR.

**Comment by guym4c** [comment]
Sorry for the delay in getting back - I think your suggestion regarding `?self` is correct, simply a typo. I'll look into the `$offset` comments and resolve this before the weekend.

**Comment by guym4c** [comment]
I've just released 2.0.0 which fixes this. A key change for your code is that `nextPage()` returns your data in a _new_ object, not modifying the existing page. Thanks for your report. :)
